src/EventsProcessor/AzureQueueHostedService.cs

```diff
@@ -109,30 +108,27 @@ public async Task ProcessQueueMessageAsync(string message, CancellationToken can
                 _logger.LogInformation("Processing message.");
                 var events = new List<IEvent>();
 
-                var token = JToken.Parse(message);
-                if (token is JArray)
+                using var jsonDocument = JsonDocument.Parse(message);
+                var root = jsonDocument.RootElement;
+                if (root.ValueKind == JsonValueKind.Array)
                 {
-                    var indexedEntities = token.ToObject<List<EventMessage>>()
+                    var indexedEntities = root.ToObject<List<EventMessage>>()
                         .SelectMany(e => EventTableEntity.IndexEvent(e));
                     events.AddRange(indexedEntities);
                 }
-                else if (token is JObject)
+                else if (root.ValueKind == JsonValueKind.Object)
```

Comment:
Same suggestion as Mail queue, maybe we should take this as the opportunity to just enqueue arrays

Reply:
I think that's a great idea. Is there any danger in doing this in one release or does it need to be staged where we start queuing as all arrays in one release and remove the object check in the next? 

Basically is there any chance a queue has items in it during a release?

Reply:
typically our queues sit at 0 since we continue working on them until they're empty. That being said, Devops needs to be aware of these changes to upgrade in the right order to ensure the queues are empty.

We could approach it your way, too where we just start enqueueing arrays only and remove the single-object behavior next release. Either is fine by me

Reply:
We _could_ go that route, OR, use a completely new queue... leaving the old one to stack up during the release, and then new items from the new queue start getting processed correctly... We can create a quick-n-dirty Azure function or Notification pipeline to transform the old queue items and re-enqueue them to the new one in the proper Array format... we can leave that in place as an infrastructure stop-gap in the meantime... that's the other option (vs. phasing the change in)

Reply:
I think phasing it in is pretty trivial so I decided to go that way.